lib/Sema/TypeCheckDecl.cpp

```diff
@@ -6393,7 +6393,8 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
         AccessScope desiredAccessScope = AccessScope::getPublic();
         switch (access) {
         case Accessibility::Private:
-          assert(ED->getDeclContext()->isModuleScopeContext() &&
```

Comment:
I'm surprised this doesn't bail out early.  We're going to a lot of effort to validate bad extensions.